irepela/homework_9/fraction.py

```diff
@@ -41,43 +41,55 @@ def convert_to_fraction(self, other):
         return result
```

Comment:
this function actually duplicates init logic, so you have conversion in 2 places
I would suggest to change this method to

def can_convert(cls, item) -> bool


and your logic will be
```
    def __lt__(self, other):
        if not can_convert(other):         # check if need to proceed
                return NotImplemented
                
        other_fraction = type(self)(other)    # conversion
        res =  self.num * other_fraction.denom < self.denom * other_fraction.num    # prepare result
        return res

```

Reply:
In this case I would need to check once more(using isInstance) if "other" is int to cast it to Fraction:
 other_fraction = type(self)(other)    # conversion
 
 So I'll have bunch isInstance check inside can_convert and one more check inside each function.
 
 Or I didn't understand you correctly?

Reply:
can_convert(other)  in each function, thats true,
but if you need to extend functionality to support new datatypes you will have to make change in 2 places:
- can_convert check
- __init__